src/data/nat/basic.lean

```diff
@@ -311,6 +326,15 @@ protected theorem div_le_div_right {n m : ℕ} (h : n ≤ m) {k : ℕ} : n / k 
 (nat.eq_zero_or_pos k).elim (λ k0, by simp [k0]) $ λ hk,
 (le_div_iff_mul_le' hk).2 $ le_trans (nat.div_mul_le_self' _ _) h
 
+def le_half_of_double_le (m n : ℕ) (h : 2*m ≤ n) : m ≤ n/2 :=
+begin
+  convert nat.div_le_div_right h,
+  exact (nat.mul_div_cancel_left m dec_trivial).symm,
+end
+
+lemma div_lt_div_to_lt (m n k : ℕ) (h : k > 0) : m / k < n / k → m < n :=
+λ h₁, by_contradiction $ λ h₂, absurd h₁ (not_lt_of_ge (nat.div_le_div_right (not_lt.1 h₂)))
```

Comment:
Drop the hypothesis `(h : k > 0)`, which isn't necessary (or used by this proof).